src/Microsoft.TemplateEngine.Abstractions/IPostAction.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Abstractions
+{
+    public interface IPostAction
+    {
+        int Order { get; }
+
+        string Name { get; }
+
+        IReadOnlyList<IPostActionOperation> Operations { get; }
+
+        IReadOnlyList<IPostActionOperation> AlternateOperations { get; }
```

Comment:
I agree with all of what you’ve mentioned here. I set up “CommandText” basically as a placeholder until we we’re more clear regarding what exactly we want. It may not be necessary if we agree that what I’m currently calling “Name” is mapped in code into exactly what we want run. What I mean is (using your example), if name = “npm” gives us everything we need to run the npm action, then CommandText is superfluous.

Let’s find a couple minutes to go over this and finzalize what’s needed.

From: Mike Lorbetske [mailto:notifications@github.com]
Sent: Tuesday, September 6, 2016 8:55 PM
To: dotnet/templating <templating@noreply.github.com>
Cc: Sean Peters <sepeters@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [dotnet/templating] Setup processing of post operation configuration (#20)


In src/Microsoft.TemplateEngine.Abstractions/IPostAction.cs<https://github.com/dotnet/templating/pull/20#discussion_r77756829>:

> @@ -0,0 +1,15 @@

> +﻿using System.Collections.Generic;

> +

> +namespace Microsoft.TemplateEngine.Abstractions

> +{

> +    public interface IPostAction

> +    {

> +        int Order { get; }

> +

> +        string Name { get; }

> +

> +        IReadOnlyList<IPostActionOperation> Operations { get; }

> +

> +        IReadOnlyList<IPostActionOperation> AlternateOperations { get; }

I think these are intended to just be user facing instructions, we should rename/retype as such. Perhaps following up with Damian for clarification on what should happen if we didn't understand the command family (name I'm guessing is the equivalent to the family of commands).

I'm also a little unclear on CommandText, does it give the flexibility for callers to understand how to do parameter substitutions or whatnot if required? I suppose they could run through the processor to replace tokens and whatnot, but giving some sort of a structured payload (maybe just key value pairs) instead and leaving it up to the thing that interprets the command family in the caller to appropriately structure the command line or determine the component in the caller (maybe it invokes something in VS/XS/etc. to take an action instead of shelling out).

My imagination on how this would be populated (without changing away from CommandText):
Order => 0
Name => NPM
Operations => [ { CommandText => "npm install" } ]
ManualInstructions (or something like that) => [ "Run 'npm install', if you don't have npm, you can get it from (place that one gets npm from)" ]

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/dotnet/templating/pull/20/files/1fab7b93d9c5049829fbc5373bcee701741a9bab#r77756829>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AT8a8XGcLsjmsMt19gWpAPuqtHublUbmks5qnjWCgaJpZM4J0KYD>.